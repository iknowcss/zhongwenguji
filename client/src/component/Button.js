import React from 'react';
import cx from 'classnames';
import style from './Button.module.scss';

export default ({ secondary, accent, ...props }) => (
  <button {...props} className={cx(style.button, {
    [style.secondary]: secondary,
    [style.accent]: accent,
  }, props.className)} />
);
