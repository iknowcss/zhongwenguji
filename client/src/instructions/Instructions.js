import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { dismissInstructions } from './instructionsActions';
import { setCharacterSetSimplified, setCharacterSetTraditional } from '../characterTest/characterTestActions';
import Button from '../component/Button';
import {
  SwipeLeftIcon,
  SwipeRightIcon,
  KeyboardArrowLeftIcon,
  KeyboardArrowRightIcon,
  SearchIcon,
  UndoIcon
} from '../component/Icon';
import I18n from '../i18n/I18n';
import noop from '../util/noop';
import style from './Instructions.module.scss'

class Instructions extends Component {
  static propTypes = {
    isTouch: PropTypes.bool,
    dismissInstructions: PropTypes.func,
    setCharacterSetSimplified: PropTypes.func,
    setCharacterSetTraditional: PropTypes.func
  };

  static defaultProps = {
    touch: true,
    dismissInstructions: noop,
    setCharacterSetSimplified: noop,
    setCharacterSetTraditional: noop,
  };

  handleCharacterSetToggle = (event) => {
    if (event.target.checked) {
      this.props.setCharacterSetTraditional();
    } else {
      this.props.setCharacterSetSimplified();
    }
  };

  handleStartClick = (event) => {
    event.preventDefault();
    this.props.dismissInstructions();
  };

  render() {
    const { touch } = this.props;
    return (
      <div className={cx(style.container, this.props.className)}>
        <section className={style.modal}>
          <I18n component="p" className={style.paragraph} stringId="instructions.overview" />

          {touch ? (<>
            <div className={style.iconParagraphContainer}>
              <SwipeRightIcon size="medium" className={style.iconParagraphIcon}/>
              <I18n component="p" className={cx(style.iconParagraph)} stringId="instructions.swipeRight" />
            </div>

            <div className={style.iconParagraphContainer}>
              <SwipeLeftIcon size="medium" className={style.iconParagraphIcon}/>
              <I18n component="p" className={style.iconParagraph} stringId="instructions.swipeLeft" />
            </div>
          </>) : (<>
            <div className={style.iconParagraphContainer}>
              <KeyboardArrowRightIcon size="small" className={cx(style.iconParagraphIcon, style.iconParagraphKeyboardIcon)}/>
              <I18n component="p" className={cx(style.iconParagraph)} stringId="instructions.keyboardRight" />
            </div>

            <div className={style.iconParagraphContainer}>
              <KeyboardArrowLeftIcon size="small" className={cx(style.iconParagraphIcon, style.iconParagraphKeyboardIcon)}/>
              <I18n component="p" className={style.iconParagraph} stringId="instructions.keyboardLeft" />
            </div>
          </>)}

          <div className={style.iconParagraphContainer}>
            <div className={style.fauxButton}>
              <SearchIcon size="tiny" className={style.iconParagraphIcon}/>
            </div>
            <I18n component="p" className={style.iconParagraph} stringId={`instructions.definition${touch ? 'Touch' : 'Mouse'}`} />
          </div>

          <div className={style.iconParagraphContainer}>
            <div className={style.fauxButton}>
              <UndoIcon size="tiny" className={style.iconParagraphIcon}/>
            </div>
            <I18n component="p" className={style.iconParagraph} stringId={`instructions.undo${touch ? 'Touch' : 'Mouse'}`} />
          </div>

          <div className={style.characterSetContainer}>
            <input type="checkbox" id="traditionalCheckbox" onChange={this.handleCharacterSetToggle}/> <label htmlFor="traditionalCheckbox">Traditional</label>
          </div>

          <I18n
            component={Button}
            className={style.button}
            stringId={'instructions.startTestButton'}
            onClick={this.handleStartClick}
          />
        </section>
      </div>
    );
  }
}

export { Instructions as Pure };

export default connect(null, {
  dismissInstructions,
  setCharacterSetSimplified,
  setCharacterSetTraditional,
})(Instructions);
