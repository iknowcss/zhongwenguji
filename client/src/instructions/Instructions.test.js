import React from 'react';
import Renderer from 'react-test-renderer';
import Button from '../component/Button';
import { Pure as Instructions } from './Instructions';

describe('Instructions', () => {
  it('renders keyboard instructions', () => {
    const component = Renderer.create(
      <Instructions />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders touch instructions', () => {
    const component = Renderer.create(
      <Instructions touch />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('dismisses instructions', () => {
    const dismissInstructions = jest.fn();
    const preventDefault = jest.fn();
    const component = Renderer.create(
      <Instructions dismissInstructions={dismissInstructions} />
    );
    const startSimplifiedButton = component.root.findAllByType(Button)[0];
    startSimplifiedButton.props.onClick({ preventDefault });

    expect(preventDefault).toHaveBeenCalledTimes(1);
    expect(dismissInstructions).toHaveBeenCalledTimes(1);
  });

  describe('toggle character set', () => {
    let component;
    let checkbox;
    let setCharacterSetTraditional;
    let setCharacterSetSimplified;

    beforeEach(() => {
      setCharacterSetTraditional = jest.fn();
      setCharacterSetSimplified = jest.fn();
      component = Renderer.create(
        <Instructions
          setCharacterSetTraditional={setCharacterSetTraditional}
          setCharacterSetSimplified={setCharacterSetSimplified}
        />
      );
      checkbox = component.root.findAllByProps({ id: 'traditionalCheckbox' })[0];
    });

    it('toggles the traditional character set', () => {
      checkbox.props.onChange({ target: { checked: true }});
      expect(setCharacterSetSimplified).not.toHaveBeenCalled();
      expect(setCharacterSetTraditional).toHaveBeenCalledTimes(1);
    });

    it('toggles the simplified character set', () => {
      checkbox.props.onChange({ target: { checked: false }});
      expect(setCharacterSetSimplified).toHaveBeenCalledTimes(1);
      expect(setCharacterSetTraditional).not.toHaveBeenCalled();
    });
  });
});
