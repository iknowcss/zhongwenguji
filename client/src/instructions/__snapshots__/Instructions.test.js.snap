// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions renders keyboard instructions 1`] = `
<div
  className="container"
>
  <section
    className="modal"
  >
    <p
      className="paragraph"
    >
      instructions.overview
    </p>
    <div
      className="iconParagraphContainer"
    >
      <span
        className="span span--medium swipeRightIcon iconParagraphIcon"
      >
        <svg
          className="svg"
          viewBox="0 0 397.233 397.233"
        >
          <use
            xlinkHref="swipeRight.svg#main"
          />
        </svg>
      </span>
      <p
        className="iconParagraph"
      >
        instructions.swipeRight
      </p>
    </div>
    <div
      className="iconParagraphContainer"
    >
      <span
        className="span span--medium swipeLeftIcon iconParagraphIcon"
      >
        <svg
          className="svg"
          viewBox="0 0 397.232 397.232"
        >
          <use
            xlinkHref="swipeLeft.svg#main"
          />
        </svg>
      </span>
      <p
        className="iconParagraph"
      >
        instructions.swipeLeft
      </p>
    </div>
    <div
      className="iconParagraphContainer"
    >
      <div
        className="fauxButton"
      >
        <span
          className="span span--tiny searchIcon iconParagraphIcon"
        >
          <svg
            className="svg"
            viewBox="0 0 250.313 250.313"
          >
            <use
              xlinkHref="search.svg#main"
            />
          </svg>
        </span>
      </div>
      <p
        className="iconParagraph"
      >
        instructions.definitionTouch
      </p>
    </div>
    <div
      className="iconParagraphContainer"
    >
      <div
        className="fauxButton"
      >
        <span
          className="span span--tiny undoIcon iconParagraphIcon"
        >
          <svg
            className="svg"
            viewBox="0 0 511.63 511.631"
          >
            <use
              xlinkHref="undo.svg#main"
            />
          </svg>
        </span>
      </div>
      <p
        className="iconParagraph"
      >
        instructions.undoTouch
      </p>
    </div>
    <div
      className="characterSetContainer"
    >
      <input
        id="traditionalCheckbox"
        onChange={[Function]}
        type="checkbox"
      />
       
      <label
        htmlFor="traditionalCheckbox"
      >
        Traditional
      </label>
    </div>
    <button
      className="button button"
      onClick={[Function]}
    >
      instructions.startTestButton
    </button>
  </section>
</div>
`;

exports[`Instructions renders touch instructions 1`] = `
<div
  className="container"
>
  <section
    className="modal"
  >
    <p
      className="paragraph"
    >
      instructions.overview
    </p>
    <div
      className="iconParagraphContainer"
    >
      <span
        className="span span--medium swipeRightIcon iconParagraphIcon"
      >
        <svg
          className="svg"
          viewBox="0 0 397.233 397.233"
        >
          <use
            xlinkHref="swipeRight.svg#main"
          />
        </svg>
      </span>
      <p
        className="iconParagraph"
      >
        instructions.swipeRight
      </p>
    </div>
    <div
      className="iconParagraphContainer"
    >
      <span
        className="span span--medium swipeLeftIcon iconParagraphIcon"
      >
        <svg
          className="svg"
          viewBox="0 0 397.232 397.232"
        >
          <use
            xlinkHref="swipeLeft.svg#main"
          />
        </svg>
      </span>
      <p
        className="iconParagraph"
      >
        instructions.swipeLeft
      </p>
    </div>
    <div
      className="iconParagraphContainer"
    >
      <div
        className="fauxButton"
      >
        <span
          className="span span--tiny searchIcon iconParagraphIcon"
        >
          <svg
            className="svg"
            viewBox="0 0 250.313 250.313"
          >
            <use
              xlinkHref="search.svg#main"
            />
          </svg>
        </span>
      </div>
      <p
        className="iconParagraph"
      >
        instructions.definitionTouch
      </p>
    </div>
    <div
      className="iconParagraphContainer"
    >
      <div
        className="fauxButton"
      >
        <span
          className="span span--tiny undoIcon iconParagraphIcon"
        >
          <svg
            className="svg"
            viewBox="0 0 511.63 511.631"
          >
            <use
              xlinkHref="undo.svg#main"
            />
          </svg>
        </span>
      </div>
      <p
        className="iconParagraph"
      >
        instructions.undoTouch
      </p>
    </div>
    <div
      className="characterSetContainer"
    >
      <input
        id="traditionalCheckbox"
        onChange={[Function]}
        type="checkbox"
      />
       
      <label
        htmlFor="traditionalCheckbox"
      >
        Traditional
      </label>
    </div>
    <button
      className="button button"
      onClick={[Function]}
    >
      instructions.startTestButton
    </button>
  </section>
</div>
`;
